File: engine/engine-model/src/main/scala/tools/packet/storage/GetStorage.scala
package tools.packet.storage

import client.inventory.Item
import net.opcodes.SendOpcode
import tools.packet.PacketInput

class GetStorage(private var _npcId: Int, private var _slots: Byte, private var _items: java.util.List[Item], private var _meso: Int) extends PacketInput {
  def npcId: Int = _npcId

  def slots: Byte = _slots

  def items: java.util.List[Item] = _items

  def meso: Int = _meso

  override def opcode(): SendOpcode = SendOpcode.STORAGE
}

File: engine/engine-model/src/main/java/tools/packet/storage/GetStorage.java
package tools.packet.storage;

import java.util.List;

import client.inventory.Item;
import net.opcodes.SendOpcode;
import tools.packet.PacketInput;

public record GetStorage(Integer npcId, Byte slots, List<Item> items, Integer meso) implements PacketInput {
   @Override
   public SendOpcode opcode() {
      return SendOpcode.STORAGE;
   }
}